File: Backend/MyApp.CoreService/DTOs/Responses/TransactionResponse.cs
using MyApp.CoreService.Enums;

namespace MyApp.CoreService.DTOs.Responses;

public record TransactionResponse(
    long Id,
    int AccountId,
    TransactionType Type,
    decimal Amount,
    decimal BalanceBefore,
    decimal BalanceAfter,
    int? RelatedAccountId,
    string? Description,
    DateTimeOffset CreatedAt
);

File: Mobile/components/account/src/main/java/nekit/corporation/loan_shared/data/datasource/remote/model/TransactionResponse.kt
package nekit.corporation.loan_shared.data.datasource.remote.model

import kotlinx.serialization.Contextual
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.OffsetDateTime

@Serializable
data class TransactionResponse(
    val id: Long,
    val accountId: Int,
    val type: TransactionType,
    val amount: Double,
    val balanceBefore: Double,
    val balanceAfter: Double,
    @SerialName("relatedAccountId") val relatedAccountId: Int? = null,
    val description: String? = null,
    @SerialName("createdAt")  @Contextual val createdAt: OffsetDateTime
)
